website/content/en/preview/testing-guide

```diff
@@ -0,0 +1,41 @@
+# Karpenter Testing User Guide
+
+Currently, users can only test Karpenter by adding to the list of integration tests run in a mock environment or installing and testing Karpenter on a real cluster. Users that want to test more are limited by the lack of automation and configurability, resulting in much less testing.
+
+To increase testing of Karpenter, we will introduce infrastructure and tools to extend the current mechanisms and enable developers to contribute with more confidence.
+
+## Mediums of Testing
+
+__Code Contributions__ are currently tested with integration tests run by GitHub Actions. Testing a contribution comprehensively is hard for a few reasons:
+
+* Testing all supported Kubernetes and Karpenter versions would be very tedious for one developer
+* Testing with real instances would incur unwanted costs to the contributor
+
+__On-demand testing__ is where a developer may need to test Karpenter independent of contribution. Testing here may only rely on some test configurations but could be for the following:
+
+* Scale tests to verify test workloads or dependencies
+* Benchmark performance of custom builds and changes
+
+__Periodic tests__ are meant to run on a pre-determined basis to monitor the health of the Karpenter project. These can be used to:
+
+* Track performance regressions/progressions over time
+* Validate that Karpenter functionally works for existing and future releases.
+
+## User Workflow
+
+__(To be implemented)__ When contributing, users will need to go through steps to automatically test their changes.
+
+* When a user cuts a PR and is ready to test, a maintainer will need to approve testing by posting `/ok-to-test`
```

Comment:
Is this optional?

Reply:
This will not be optional.
> a maintainer will need

A maintainer will be able to do `/ok-to-test` on their own PRs.